File: Data Structures & Algorithms/reverse-integer/submission-0.js
class Solution {
    /**
     * @param {number} x
     * @return {number}
     */
    reverse(x) {
        let ref = x;
        let out = 0;

    if(x < 0)
      {  ref *= (-1);}

    while (ref > 0)
    {
        let temp = ~~(ref % 10);
        out *= 10;
        out += temp;
        ref = ~~(ref / 10)
    }
    if(out >2147483647)
        return 0;
    return (x < 0)? out *= (-1): out;
    }

}

File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s)
{
	if (s.empty())
		return 1;
	vector<int> ref(s.length());
	bool Palindrome = true;
	regex pattern("^[a-z0-9]$");
	int vec_i = 0;
	for (int i = 0; i < (int)s.length(); i++)
	{
		if (s[i] <= 'Z' && s[i] >= 'A')
		{
			s[i] = tolower(s[i]);
		}
		string temp(1, s[i]);
		if (regex_match(temp, pattern))
		{
			ref[vec_i] = (int)s[i];
			vec_i++;
		}
	}

	ref.resize(vec_i);
	for (int i = 0, j = (int)ref.size() - 1; i < ((int)ref.size() / 2); i++, j--)
	{
		if (ref[i] != ref[j])
		{
			Palindrome = false;
		}
	}

	return Palindrome;
}
};
